core/src/main/java/org/apache/spark/util/collection/unsafe/sort/PrefixComparators.java

```diff
@@ -141,7 +141,7 @@ public int compare(long bPrefix, long aPrefix) {
     @Override public boolean sortSigned() { return true; }
     @Override public boolean nullsFirst() { return true; }
     public int compare(long a, long b) {
-      return (a < b) ? -1 : (a > b) ? 1 : 0;
+      return Long.compare(a, b);
```

Comment:
```
    /**
     * Compares two {@code long} values numerically.
     * The value returned is identical to what would be returned by:
     * <pre>
     *    Long.valueOf(x).compareTo(Long.valueOf(y))
     * </pre>
     *
     * @param  x the first {@code long} to compare
     * @param  y the second {@code long} to compare
     * @return the value {@code 0} if {@code x == y};
     *         a value less than {@code 0} if {@code x < y}; and
     *         a value greater than {@code 0} if {@code x > y}
     * @since 1.7
     */
    public static int compare(long x, long y) {
        return (x < y) ? -1 : ((x == y) ? 0 : 1);
    }
```